dlt/common/normalizers/json/relational.py

```diff
@@ -79,21 +80,27 @@ def _is_complex_type(self, table_name: str, field_name: str, _r_lvl: int) -> boo
         # turn everything at the recursion level into complex type
         max_nesting = self.max_nesting
         schema = self.schema
+        table = schema.tables.get(table_name)
+        if table:
```

Comment:
are we able to cache nesting per table in `extend_table`? and here to only access a dict with a cache? why: because this is called for each dict/list in the results so is time critical

Reply:
Also I am not sure if I understand `extend_table` ’s purpose and how to use it, grepping in source code doesn’t give definitive examples I could use, should I just define some dictionary for this? Now that we have getter and setter in a resource we will have to invalidate this as well.

Reply:
@rudolfix can you please take a look again?